cmd/upgrade/app/v1alpha1/result.go

```diff
@@ -0,0 +1,258 @@
+/*
+Copyright 2019 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"math/rand"
+	"time"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	upgrade "github.com/openebs/maya/pkg/apis/openebs.io/upgrade/v1alpha1"
+	stringer "github.com/openebs/maya/pkg/apis/stringer/v1alpha1"
+	objectmeta "github.com/openebs/maya/pkg/kubernetes/objectmeta/v1alpha1"
+	ownerreference "github.com/openebs/maya/pkg/kubernetes/ownerreference/v1alpha1"
+	typemeta "github.com/openebs/maya/pkg/kubernetes/typemeta/v1alpha1"
+	upgraderesult "github.com/openebs/maya/pkg/upgrade/result/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	jobNameLabelKey       = "openebs.io/upgradejob"
+	itemNameLabelKey      = "openebs.io/upgradeitemname"
+	itemNamespaceLabelKey = "openebs.io/upgradeitemnamespace"
+	itemKindLabelKey      = "openebs.io/upgradeitemkind"
+)
+
+// UpgradeResult is a wrapper over upgrade.UpgradeResult struct
+type UpgradeResult struct {
+	object *upgrade.UpgradeResult
+}
+
+// UpgradeResultBuilder helps to build UpgradeResult instance
+type UpgradeResultBuilder struct {
+	InstanceName      string
```

Comment:
Was there any issue following the usual pattern of composing a UpgradeResult object. e.g.
```go
type UpgradeResultBuilder struct {
  object *apis.UpgradeResult
  errors []error
}
```

Reply:
Should this be defined in `pkg/upgrade/v1alpha1/upgraderesult.go`
I guess core packaging for upgrade is `pkg/upgrade/v1alpha1`

Reply:
This is not unit of upgrade. This is required to set owner reference.
IMO It is ok to have it here.

Reply:
May be we can rename it to `SelfName`,`SelfNameSpace`,`SelfUID`

Reply:
will rename it.